### PR TITLE
Add documentation for StyleSheet API

### PR DESCRIPTION
**Pull request by sgny**
Also added the relevant link to `Style.md`

### CODE REVIEW

File: reason-react-native/src/apis/StyleSheet.md [hunk 1]
```diff
@@ -1,20 +1,104 @@
...
+the key.
+
+```reason
+create: Js.t('a) => Js.t('a)
+```
+
+### `flatten`
+
+Takes an array of styles (of type `Style.t`) and returns a single style (also of
+type `Style.t`).
```
Comment (cknitt) [inline reason-react-native/src/apis/StyleSheet.md]: Maybe add a comment that using styles created with `Style.array` in `StyleSheet.create` will lead to a runtime error, so `StyleSheet.flatten` *must* be used in that case. (`Style.array` is fine for passing styles to component props though.)

File: reason-react-native/src/apis/StyleSheet.md [hunk 2]
```diff
@@ -1,20 +1,104 @@
...
+operator when fields are not explicitly set, you may use the `flatten` method
+instead.
+
+```reason
+absoluteFill: Style.t
+```
+
+### `absoluteFillObject`
+
+This is identical to `absoluteFill` when used with pure functions.
```
Comment (MoOx) [inline reason-react-native/src/apis/StyleSheet.md]: I won't say that as this thing "identical" as it's not. See #523

File: reason-react-native/src/apis/StyleSheet.md [hunk 3]
```diff
@@ -1,20 +1,104 @@
...
+
+### `create`
+
+Takes and returns `Js.t('a)` objects, with `string` keys and `Style.t` values.
+Individual styles can be accessed using `##`, followed by the name specified as
+the key.
+
+```reason
+create: Js.t('a) => Js.t('a)
+```
```
Comment (MoOx) [inline reason-react-native/src/apis/StyleSheet.md]: Same question as https://github.com/reasonml-community/bs-react-native/pull/519#discussion_r289291691

Comment (sgny) [reply reason-react-native/src/apis/StyleSheet.md]: Perhaps we revisit this issue later, but I don't think there is a way to exclude `array(Style.t)` without a lot of pain elsewhere.

File: reason-react-native/src/apis/StyleSheet.md (hunk 1 above)
Comment (sgny) [reply reason-react-native/src/apis/StyleSheet.md]: Added such comments

**Comment by MoOx** [comment]
👍

**Pull request merged**